docs/nddata/index.rst

```diff
@@ -156,6 +157,57 @@ Code that uses the arithemtic methods that used to be included in
 instead subclass `~astropy.nddata.NDDataArray`; that class is equivalent to
 the original `~astropy.nddata.NDData` class.
 
+.. _nddata_transition2:
+
+Transition to astropy x.xx
+==========================
+
+The implementation of `APE 7`_ where `~astropy.nddata.NDData` underwent a major
+revision had some bugs and minor shortcomings;
+So the interface was changed in some small ways and some bugs were fixed:
+
+`~astropy.nddata.NDDataBase`:
+
++ `NDDataBase.uncertainty.getter`: is now abstract.
++ `NDDataBase.uncertainty.setter`: is moved to `~astropy.nddata.NDData`.
+
+`~astropy.nddata.NDData`:
+
++ ``NDData`` now implements the ``uncertainty`` getter and setter from the
+  former `~astropy.nddata.NDDataBase` with some minor modifications.
+
++ ``NDData.uncertainty.setter``:
+
+  + wraps an ``uncertainty`` inside an `~astropy.nddata.UnknownUncertainty` if
+    no ``uncertainty_type`` attribute is present in the uncertainty.
+  + the uncertainty_type of the uncertainty must not be a string
+    anymore. But it is still recommended.
+  + did not set the ``parent_nddata`` of the ``uncertainty`` if the uncertainty
+    was a subclass of `~astropy.nddata.NDUncertainty`. This is now fixed.
+
++ ``NDData.__init__``:
+
+  + got an additional optional parameter ``copy`` which is
+    by default False. If it is True all other parameters are copied before saving
+    them as attributes. If False the parameters are only copied if there is no
+    way of saving them as reference.
+  + the ``data`` parameter allows now also ``Masked_Quantity`` objects
+  + if the ``data`` is another instance or subclass of
+    `~astropy.nddata.NDData` all implicit properties are checked because
+    subclasses might implement another set of restrictions.
+  + if an implicit and explicit attribute is provided, the
```

Comment:
This point should be added either as an example in getting started (show an example of what happens if you, say, already have a unit but supply a new one when creating a new NDData) or in `nddata.rst`